core/shared/src/main/scala/zio/CancelableFuture.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.concurrent.{ CanAwait, ExecutionContext, Future }
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+private[zio] abstract class CancelableFuture[+E, +A](val future: Future[A])
```

Comment:
We might want to make it public so that users can use the additional cancellation ability!